apps/server/src/modules/tool/uc/school-external-tool.uc.ts

```diff
@@ -71,6 +72,26 @@ export class SchoolExternalToolUc {
 		return schoolExternalTool;
 	}
 
+	async updateSchoolExternalTool(
+		userId: EntityId,
+		schoolExternalToolId: string,
+		schoolExternalTool: SchoolExternalTool
+	): Promise<SchoolExternalToolDO> {
+		await this.ensureSchoolExternalToolPermission(userId, schoolExternalToolId);
+		await this.schoolExternalToolValidationService.validateUpdate(schoolExternalToolId, schoolExternalTool);
+
+		const loaded: SchoolExternalToolDO = await this.schoolExternalToolService.getSchoolExternalToolById(
+			schoolExternalToolId
+		);
+		const updated: SchoolExternalToolDO = new SchoolExternalToolDO({
+			...loaded,
```

Comment:
i am not sure about the spread operator with the `loaded` here since it is a put and should always override the current entity. I think `new SchoolExternalToolDO({ id: schoolExternalToolId, ...schoolExternalTool })` is enough.